### ISSUE TITLE
Relocate Image Group

### ISSUE THREAD

**Issue by edickens** [FreeUKGen/MyopicVicar#1447]
Priority (0 3 8 4) 15

**Comment by edickens** [comment]
Now that we are getting some live experience of using the image software, I suggest we need a "Relocate Image Group" in the same way, and for much the same reasons, that we needed a "Relocate Batch".
The relocate needs to offer the same sort of options as does the relocate batch, but would include Image Source and Image Group.  No new Church. Register Type or Image Source will need to be created.  This should have been done in advance.  The Relocate should select these entries and if they are not there, the relocate cannot go ahead.  It would be necessary to be able to edit the Image Group name.
The image filenames do not need to change.
I don't know how difficult this is to do, because it will also be necessary to move the images by cut and paste on the image server.  The destination folder will have been created by the creation of the destination Image Source.
Thoughts please.  Eric

**Comment by Captainkirkdawson** [comment]
Not sure that I understand this at all

Kirk Dawson
5220 Riverside Drive
Fairmont Hot Springs, B.C.
V0B 1L1

**Comment by edickens** [comment]
Which bit do you not understand?  I'm suggesting a "relocate image group" similar to the "relocate batch".

**Comment by edickens** [comment]
Here is an example:-
![Capture.JPG](https://waffleio-direct-uploads-production.s3.amazonaws.com/uploads/542531935e93a002006d0247/125516c66e82c728ace21e0d46cb958a379791accda5fd53b652fa912a5265296b51f02e7384e671e6162351580d4bac43585f45ffac8928f0a42062c80d55b98e30063e937f03a6aa801db500bf228e4882b1b88249c4e13c37aa.JPG)
These images were all in "Whitehaven" and so were all in one Image Source.  The only way I can divide them up is to download them (fortunately I can use BitKinex and do it in one go) then upload to the new destination, then delete each image individually.
This also happens when the images are reference to the wrong Church, say "Parish Church" instead of "St Mary", and the initial upload did not know a lot of the dedications so there are quite a few of these.

**Comment by edickens** [comment]
Some clarification please.
The Move images now looks like this:-
![Capture.JPG](https://waffleio-direct-uploads-production.s3.amazonaws.com/uploads/542531935e93a002006d0247/125516c66e82c728ace21e0d46cb958a379791accda5fd53b652fa912a5265296b51f02e7384e671e6162351580d4bac43585f45ffac8928f0a42062c80d55b98e30063e937f03a6aa8318b500bf228c4b87b2b98945c9ea3c37aa.JPG)
where the options to receive the images are all Image Groups within this Source.
But we need to move the images to a different Source, usually, but not exclusively, a different Register Type in the same Church.
I suggest that the receiving Image Group must have been created before the move, so the list of destinations will still be Image Groups.  But before this it will be necessary to select a County/Place/Church/Register Type/Source and the relevant image groups displayed.  The default should be the present Source.
The major problem I see with this change is that the Images themselves need to be moved on the Image Server into a different Folder that will have been created.  The only time that the move is not needed is if the images are to stay within the same Source.
This facility will be used a lot because there are images which are under PR that should have been OT.
EricD

**Comment by bev-fc** [comment]
@edickens I think if a new folder will be created/renamed for those images, the cron script will picked them up and added into FreeReg server, then delete old image groups in FreeReg would clean it up, please correct me if I am wrong @Captainkirkdawson

**Comment by edickens** [comment]
I don't know the full details of "Cron script", butmy suggestion is that all destination Source and Image Group must be set up before the move can take place.  So the move would only involve the references ot the images and the images themselves need to be "cut" and "pasted" from the old folder on the images server to the new one which will have been created when the Source entry was made.  Only if the move is within the same Source would there be no need to move the images on the image server.
I can't see how a script can be created again.
I am not suggesting any renaming of anything.  If a mistake has been made, then the correct version has to be created and the images moved to it, after which the wrong entry is deleted by the Coordinator.

**Comment by bev-fc** [comment]
As my understanding, there is a cron script that runs everyday to add the new added images on image server into FreeReg database, based on folder name to decide which place/church/register/source/image_group it belongs based on predefined rules. 

when you move images from image_group_1 of source_1 to image_group_2 of source_2 on FreeReg, there will be no problem

when you move images from folder_1 to folder_2 on Image Server, there will be no problem neither 

but there is a possibility, when the cron job runs, it will translate the new folder_2 name into a place/church/register/source/image_group names, if any of them does not exist, the script will try to create one and insert into FreeReg database, this will lead to a duplicate record of those images. 

If the new folder name can be for sure matches the expected place/church/register/source/image_group name you created in FreeReg database, then there is no problem.

**Comment by Captainkirkdawson** [comment]
@bev-fc  There is no such cron script.

**Comment by edickens** [comment]
@bev-fc  As Kirk says, there is no such script now.  We ran that process to set up the system on the website, but it will never be run again.  It was never run every night.
What you may be thinking of is a script which works the other way round.  That is to do with UserIDs and keeps the image server UserID folders in line with any new or amended website UserIDs.  It has nothing to do with Places.  I have requested in #1466 that this stops, as every new volunteer will now use your website system.

**Comment by bev-fc** [comment]
I think I am confused now how did the new images get added into FreeReg database after the first run, but I will start to modify the code as long as this is a safe to go. Is it okay for you if I list out all image groups under one church or under one place in move page other than select county -> place -> church -> register -> source -> image group? this would be an easier and faster change from the code side.

**Comment by edickens** [comment]
After the first run, all new images have been uploaded directly into the new system.  I will never upload any more images directly into the image server not create any new folders.  I am the only person with that sort of access to the image server, so this should never happen.  By doing everything from now on through the new website system, everything should be properly indexed and managed.  We have been uploading new images and also deleting some, but all through the new website system.
I think you can list out all Image Groups under a County -> Place.  So no need to ask for any other information.  Please make the present County -> Place the default for their selection.  When you list which Image Group to select, then you will need to show the full path from County down to Image Group.

**Comment by bev-fc** [comment]
I was referring it will only list image groups under current place, how frequent it will happen that you will need to move images from one place to another place, or from one county to another county?

**Comment by edickens** [comment]
The most likely is from Register Type to Register type..  But we do move from County to County.  This is because a Place may now be in a different county when the boundaries change.  So we must allow for County change.  But never Country.

**Comment by Captainkirkdawson** [comment]
@edickens 
I am having great difficulty understanding how we could possibly implement such a move capability WITH appropriate renaming for images loaded into the database from the old system. 
Surely newly created sources and uploaded images have all been done to the correct county, place church and register and should NOT need to be moved.
Let us take an example (I was going to use your example but see you have done some restructuring)
Lets look at  CUL Workington PR 

This is a source for the Parish Register for St Michael in Workington of Cumberland

It has the following image server groups
Workington PR/CUL-Workington_Presbyterian-1745-1772-2025	
Workington PR/CUL-Workington_Presbyterian-1764-1783-2267	
Workington PR/CUL-Workington_Presbyterian-1784-1827-0065	
Workington PR/CUL-Workington_Presbyterian-1827-1837-1881	
Workington PR/CUL-Workington_PriestgateLowMeetingHouse-Ind-1737-1837-209
Workington PR/Workingto	

All the images are in the same folder Workington PR  in the county folder CUL

They have names of the image server group less the folder name with -xx.jpg in the case of the first 5 groups appended at the end.  (eg CUL-Workington_Presbyterian-1764-1783-2267-20.jpg) In the last case they have nxxxx.jpg appended (eg  Workington0005.jpg)

First comment  to be made is that none of the images are actually parish register images for St Micheal. There are 4 image groups for Presbyterian, 1 for  PriestgateLowMeetingHouse-Ind and 1 group for a Transcript of marriages for the parish register.

Let us assume that we wish to place them in the correct institution and register type. ie NO change of place or county. 
Lets do the Transcript first. We have to go to the Transcript register for St Michael, creating one if it does not exist. 
We then need to add an image server source for the register. We create that and a folder Workington St Michael TR is created on the image server.  
We can then create 1 or more image server groups for the Workington St Michael TR.
So far so good.
Now there are no images in the folder on the image server. So we have to go to the image server group that contains the images. Workington PR/Workingto	
There we can list the images for the existing group. 
So the requirement is to be able to select a set of images and MOVE them to the new location. If the MOVE is restricted to the same place and county  then we have to offer first the churches for the place and having selected one of those select the register type. The MOVE in the database as also to be reflected with a request to the image server to do the physical MOVE. In this case there is no naming conflicts.

If we permit change of place and or county then there are image name change requirements as well ie CUL-Workington_Presbyterian-1764-1783-2267-20.jpg is no longer appropriate if County and/or Place or Church or date range is changed.

If we change names then what is on the server is different from what was used to create the image in the first place and therefore now different from what the coordinator may have.

I see lots of room for problems

**Comment by edickens** [comment]
OK.  Point taken.  The image filename could be confusing if the MOVE is to a different place and the place name is in the image filename.
So how about restricting the MOVE to within a Place, or even within a Church.  These will be the most likely needs.  Any higher level will require downloading and renaming of the image filenames if thought necessary, although the format of the image filenames are not mandatory.  Some are just called 1.jpg.
The MOVE needs to be from just one image group at a time.

**Comment by edickens** [comment]
PS.  The image filename does not need to be changed, but it must not already exist in the destination.

**Comment by richpomfret** [comment]
Further discussion required.

**Comment by Captainkirkdawson** [comment]
This is required only for images fro the original image server that were incorrectly specified in the first instance. It  should not be required for images being uploaded directly now. There is a work around.

**Comment by edickens** [comment]
Steve and I are discussing this to see how important it is.  But I did one correction and the workaround took nearly two hours of error prone work.

**Comment by Captainkirkdawson** [comment]
Bev has written some code but it is incomplete until I write the physical move from one folder to another. Since this is still incoming I do not propose to start that. There are many other bugs and issues of higher priority

**Comment by edickens** [comment]
Agreed.  When Steve and I have agreed the need for this, we will get back to you.  Don't do any more work for now.

**Comment by SteveBiggs** [comment]
I think this should be restricted to moving image groups between registers in the same church. This is the only way I use the Relocate Batches action.

**Comment by edickens** [comment]
And also between Sources in the same Church.  With this, I agree with Steve.

**Comment by Captainkirkdawson** [comment]
Based on that there is  a requirement to move images between folders on the image server. There is also a requirement to change the image file name to make it consistent with the new folder name ( or at least not inconsistent with the folder name)
For example an image moved from Workington_Presbyterian-1764-1783-2267-20.jpg  to Workington St Michael TR Baptisms 1700-1800 would need to have Workington_Presbyterian-1764-1783-2267- removed

**Comment by edickens** [comment]
Yes, the folder name on the image server will need to be changed and images moved.  But we are not asking for moves between Churches as in Kirk's example above.  So I suggest that the image filenames do not need to be changed.  We are saying that only moves within the same Church are allowed and the Register Type is not in the filename.
Most of these moves are because IGs are under the wrong Register Type, particularly with old images where the Register Type is Unspecified.

**Comment by SteveBiggs** [comment]
Eric's right that we are only asking for moves between registers in the same church, but the register type IS sometimes included in the group and image name, e.g. 

![image](https://user-images.githubusercontent.com/25303607/41590521-16459bd8-73af-11e8-9f6a-1cdd1f2dcc9a.png)

Having said that, it is highly unlikely that we would want to move a Transcript group to a Parish Register or similar.

**Comment by edickens** [comment]
And if the Register Type is in the image filename, it is probably in the right place and does not need to be moved.  It is very rare for the Register Type to be in the image filename.

**Comment by Captainkirkdawson** [comment]
OK based on your reaction will do the physical move but leave the name
unchanged.

Kirk Dawson
5220 Riverside Drive
Fairmont Hot Springs, B.C.
V0B 1L1

**Comment by edickens** [comment]
When we come to testing, we have a genuine example:-
![Capture.JPG](https://waffleio-direct-uploads-production.s3.amazonaws.com/uploads/542531935e93a002006d0247/125516c66e82c728ace21e0d46cb958a379791accda5fd53b652fa912a5265296b51f02e7384e671e6162351580d4bac43585f45ffac8928f0a42062c80d55b98e30063e937f03a6aa8317b500bf22824980b0bb8b45c3eb3d37aa.JPG)
Tickhill "Unknown" should be "Other Register".
But there are IGs and images:-
![Capture1.JPG](https://waffleio-direct-uploads-production.s3.amazonaws.com/uploads/542531935e93a002006d0247/125516c66e82c728ace21e0d46cb958a379791acdeabf301b44fff9e2c4533317346f6787586e22de61b2f595f581ca9415a5f44fdfadc2ba5e02e6c981e5ab38532022d8c6853beb2901eb70bbb25894683b3ba824ac8eb3d27f5c6.JPG)
The image filenames are like this:-
WRY-Tickhill_St.Mary-BU-1538-1904-A-Doyle_Bowser-Bradford.jpg
so do not need changing.
On the Image SErver there is just one folder:-
![Capture2.JPG](https://waffleio-direct-uploads-production.s3.amazonaws.com/uploads/542531935e93a002006d0247/125516c66e82c728ace21e0d46cb958a379791acddabf301b44fff9e2c4533317346f6787586e22de61b2f595f581ca9415a5f44fdfadc2ba5e02e6c981e5ab38532022d8c6853beb2901eb60bbb25894683b3bb8a45c4ed3f23f5c6.JPG)
So it looks as if the image have been moved as they are in OT?  ED

**Comment by edickens** [comment]
Do the programmers have the information they need?  This is needed as soon as possible.
Thanks
EricD

**Comment by Captainkirkdawson** [comment]
Yes but story is still in incoming and not flagged as urgent

**Comment by Captainkirkdawson** [comment]
Unsure of the purpose of the last comment or how it is applicable to the story

**Comment by Sherlock21** [comment]
My involvement arises because I tried to correct an error in the Register Type in a batch, using the tools I ahem used for years.
Only to find I have now created an empty row in the table for Tickhill - which I can't get rid of due to the inability to untangle the images. The result looks bad. - Tidyness has been one of my objectives from the start, and clean up as we go along has been my goal since day one.

A tatty display does not signify an organised setup i my book. i.e. no credibility.
Thats how  my comment fits into the story
